src/mscorlib/src/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -218,6 +258,126 @@ internal unsafe int IndexOfCore(string source, string target, int startIndex, in
             }
         }
 
+        internal unsafe int IndexOfCore(ReadOnlySpan<char> source, ReadOnlySpan<char> target, CompareOptions options, int* matchLengthPtr)
+        {
+            Debug.Assert(!_invariantMode);
+            Debug.Assert(source.Length != 0);
+            Debug.Assert(target.Length != 0);
+            Debug.Assert((options == CompareOptions.None || options == CompareOptions.IgnoreCase));
```

Comment:
> Debug.Assert((options == CompareOptions.None || options == CompareOptions.IgnoreCase)); [](start = 11, length = 88)

Although this may look true from the calls coming from Span but we may support spans in CompareInfo which will allow passing more options than what we are currently passing. so we'll need to re-visit this code again if we did that in the future. I prefer writing this code here with the assumptions you may get more options than what we are currently passing from the span APIs.

Reply:
What options should this Debug.Assert check then?

Reply:
remove the assert. 

---
In reply to: [169814071](https://github.com/dotnet/coreclr/pull/16467#discussion_r169814071) [](ancestors = 169814071)

Reply:
We have similar assert here - https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Globalization/CompareInfo.Unix.cs#L180